lib/dictBuilder/cover.c

```diff
@@ -729,11 +729,11 @@ ZDICTLIB_API size_t ZDICT_trainFromBuffer_cover(
   /* Checks */
   if (!COVER_checkParameters(parameters, dictBufferCapacity)) {
     DISPLAYLEVEL(1, "Cover parameters incorrect\n");
-    return ERROR(GENERIC);
+    return ERROR(parameter_unsupported);
```

Comment:
I think `parameter_outOfBound` is a better error code, here and elsewhere.

Generally `parameter_unsupported` is meant to mean zstd doesn't understand the parameter, but here its the parameter value that is bad.